src/yoti_common/anchor.processor.js

```diff
@@ -26,24 +26,77 @@ class AnchorProcessor {
    *
    * @param anchors
    *
-   * @returns {YotiAnchor[]}
+   * @returns {Object.<string, YotiAnchor[]>}
    */
   static process(anchors) {
     // This will contain a list of anchor values as objects
-    let anchorsData = this.getResultFormat();
+    const anchorsData = this.getResultFormat();
     for (let i = 0; i < anchors.length; i += 1) {
-      const yotiAnchorsList = this.processSingleAnchor(anchors[i]);
-      anchorsData = this.mergeAnchorsLists(anchorsData, yotiAnchorsList);
+      const certificatesList = anchors[i].originServerCerts;
+      const signedTimestamp = this.processSignedTimeStamp(anchors[i].getSignedTimeStamp());
+      const originServerCerts = AnchorProcessor.convertCertsListToX509(certificatesList);
+      const subType = anchors[i].getSubType();
+
+      const yotiAnchor = this.getAnchorFromCerts(
+        certificatesList,
+        signedTimestamp,
+        originServerCerts,
+        subType
+      );
+
+      const key = this.getAnchorListKeyByType(yotiAnchor.getType());
+      anchorsData[key].push(yotiAnchor);
     }
     return anchorsData;
   }
 
+  /**
+   * Get an anchor from certificate list.
+   *
+   * @param {Array} certificatesList
+   * @param {YotiSignedTimeStamp} signedTimestamp
+   * @param {Certificate[]} originServerCerts
+   * @param {string} subType
+   *
+   * @returns {YotiAnchor}
+   */
+  static getAnchorFromCerts(certificatesList, signedTimestamp, originServerCerts, subType) {
+    let anchor = null;
+
+    certificatesList.forEach((certificate) => {
+      const extensionsData = AnchorProcessor.convertCertToX509(certificate).extensions;
+
+      Object.keys(ANCHOR_TYPES).forEach((type) => {
+        const oid = ANCHOR_TYPES[type];
+        const extension = this.getExtensionByOid(extensionsData, oid);
+
+        if (extension !== null) {
+          anchor = new YotiAnchor(
```

Comment:
Not sure if I'm missing something, but is there a reason why we're not just doing `return new YotiAnchor(...` here, rather than assigning that to `anchor` and then returning `anchor` below?

Reply:
I've changed this in 5d285b0 - I agree returning early here is clearer and inline with other SDKs